itests/hive-unit/src/test/java/org/apache/hive/beeline/TestHplSqlViaBeeLine.java

```diff
@@ -1411,8 +1404,174 @@ private void testScriptFile(String scriptText, List<String> argList, String expe
+      }
+
+      @Override
+      public Pattern getPattern() {
+        return PATTERN;
+      }
+    };
+
+    /** Supports dates such as '2025-12-02' */
+    TimeCheckHelper DATE = new TimeCheckHelper() {
+      Pattern PATTERN = Pattern.compile("[0-9]{4}-[0-9]{1,2}-[0-9]{1,2}");
+
+      @Override
+      public long getCurrentTime() {
+        Date date = new Date(System.currentTimeMillis());
+        return strToTime(date.toString());
+      }
+
+      @Override
+      public long strToTime(String str) {
+        return Date.valueOf(str).getTime();
+      }
+
+      @Override
+      public Pattern getPattern() {
+        return PATTERN;
+      }
+    };
+
+    /** Supports datetimes such as '2025-12-02 02:56:08' */
+    TimeCheckHelper DATETIME = new TimeCheckHelper() {
+      Pattern PATTERN = Pattern.compile("[0-9]{4}-[0-9]{1,2}-[0-9]{1,2} +[0-9]{2}:[0-9]{2}:[0-9]{2}");
+
+      @Override
+      public long getCurrentTime() {
+        return System.currentTimeMillis() / 1000;
+      }
+
+      @Override
+      public long strToTime(String str) {
+        return Timestamp.valueOf(str).getTime() / 1000;
+      }
+
+      @Override
+      public Pattern getPattern() {
+        return PATTERN;
+      }
+    };
+
+    /** The current time. The resolution (e.g., milliseconds) must be the same as {@link #strToTime(String)}. */
+    long getCurrentTime();
+
+    /**
+     * Converts a candidate match to a time. The resolution (e.g., milliseconds) must be the same
+     * as {@link #getCurrentTime()}. */
+    long strToTime(String str);
+
+    /** A pattern to find string representations of the time in the output. */
+    Pattern getPattern();
+
+    /** Limits the candidates to the ones that match prefixPattern followed by this pattern. */
+    default TimeCheckHelper withPrefixPattern(Pattern prefixPattern) {
+      TimeCheckHelper parent = this;
+      return new TimeCheckHelper() {
+        @Override
+        public long getCurrentTime() {
+          return parent.getCurrentTime();
+        }
+
+        @Override
+        public long strToTime(String str) {
+          Matcher m = parent.getPattern().matcher(str);
+          if (!m.find(0)) {
+            throw new IllegalStateException("String does not contain a match for the pattern " + parent.getPattern()
+                .pattern() + ":\nThe string was: " + str);
+          }
+          return parent.strToTime(str.substring(m.start()));
+        }
+
+        @Override
+        public Pattern getPattern() {
+          return Pattern.compile(prefixPattern + parent.getPattern().pattern());
+        }
+
+        @Override
+        public TimeCheckHelper withPrefixPattern(Pattern prefixPattern) {
+          throw new UnsupportedOperationException();
+        }
+      };
+    }
+  }
+
+  /** Checks that the output of the script mentions a time as defined by the helper. */
+  private void testTimeScriptFile(String scriptText, List<String> argList, TestBeeLineWithArgs.OutStream outStream,
```

Comment:
I think it would nicer to have this as an overload:
```java
private void testScriptFile(String scriptText, List<String> argList, TimeCheckHelper checker)
```

Reply:
Fixed.